src/command_modules/azure-cli-network/azure/cli/command_modules/network/custom.py

```diff
@@ -1884,8 +1884,20 @@ def import_zone(resource_group_name, zone_name, file_name):
                 record = _build_record(entry)
                 record_set = record_sets.get(record_set_key, None)
                 if not record_set:
+
+                    # Workaround for issue #2824
+                    relative_record_set_name = record_set_name.rstrip('.')
+                    if not relative_record_set_name.endswith(origin):
+                        logger.warning(
+                            'Cannot import %s. Only records relative to origin may be '
+                            'imported at this time. Skipping...', relative_record_set_name)
+                        continue
+
+                    if record_set_type != 'soa' and relative_record_set_name != origin:
+                        relative_record_set_name = record_set_name[:-(len(origin) + 2)]
```

Comment:
Part of the workaround to strip the origin back off of the name. The old implementation was to use absolute names. If/when the service fixes itself to allow this, I will just remove this workaround and import all records as absolute references (which the DNS import spec says it should support).